eslint/.eslintrc.base.json

```diff
@@ -0,0 +1,225 @@
+{
+  "env": {
+    "browser": true,
+    "node": true,
+    "es6": true
+  },
+  "parserOptions": {
+    "ecmaVersion": 2020,
+    "sourceType": "module",
+    "ecmaFeatures": {
+      "jsx": true
+    }
+  },
+  "plugins": [
+    "import",
+    "react",
+    "react-hooks",
+    "react-memo",
+    "jsdoc",
+    "eslint-plugin-no-cyrillic-string"
+  ],
+  "extends": [
+    "eslint:recommended",
+    "plugin:react/recommended",
+    "plugin:jsx-a11y/recommended",
+    "plugin:prettier/recommended",
+    "plugin:jsdoc/recommended",
+    "plugin:import/recommended"
+  ],
+  "settings": {
+    "react": {
+      "version": "detect"
+    },
+    "import/resolver": {
+      "node": {
+        "extensions": [".ts", ".tsx", ".js", ".jsx"],
+        "moduleDirectory": ["node_modules", "src"]
+      }
+    },
+    "import/internal-regex": "^@freedom-fi/"
+  },
+  "rules": {
+    // BASIC RULES
+    "curly": ["error", "all"],
+    "prefer-template": "error",
+    "radix": "error",
+    "no-console": "error",
+    "no-empty-function": "error",
+    "no-nested-ternary": "error",
+    "no-param-reassign": "error",
+    "no-shadow": "error",
+    "no-redeclare": "error",
+    "no-restricted-imports": [
+      "error",
+      {
+        "paths": [
+          {
+            "name": ".",
+            "message": "Usage of imports from root index file of the module is not allowed."
+          }
+        ],
+        "patterns": [
+          {
+            "group": ["../*"],
+            "message": "Usage of relative parent imports is not allowed."
+          },
+          {
```

Comment:
why do we need so many custom rules?

Reply:
If it is about the whole setup, this is what I came up with in the previous projects. If about the selected line, this is about restricting relative imports. But generally, I can explain why every rule is here)